data/json/recipes/food/frozen.json

```diff
@@ -0,0 +1,260 @@
+[
+  {
+    "result": "milkshake",
+    "type": "recipe",
+    "category": "CC_FOOD",
+    "subcategory": "CSC_FOOD_OTHER",
+    "skill_used": "cooking",
+    "difficulty": 1,
+    "time": 10000,
+    "autolearn": true,
+    "qualities": [ { "id": "COOK", "level": 3 } ],
+    "tools": [ [ [ "rock_quern", -1 ], [ "clay_quern", -1 ], [ "mortar_pestle", -1 ] ], [ [ "food_processor", 10 ] ] ],
+    "components": [
+      [ [ "milk", 1 ] ],
+      [ [ "con_milk", 1 ], [ "can_coconut", 1 ] ],
+      [
+        [ "chocolate", 1 ],
+        [ "candy", 1 ],
+        [ "candy2", 1 ],
+        [ "candy3", 1 ],
+        [ "maple_candy", 1 ],
+        [ "syrup", 1 ],
+        [ "strawberries", 1 ],
+        [ "strawberries_cooked", 1 ],
+        [ "irradiated_strawberries", 1 ],
+        [ "blueberries", 1 ],
+        [ "blueberries_cooked", 1 ],
+        [ "irradiated_blueberries", 1 ],
+        [ "blackberries", 1 ],
+        [ "irradiated_blackberries", 1 ],
+        [ "cherries", 1 ],
+        [ "irradiated_cherries", 1 ]
+      ],
+      [ [ "sugar", 10 ] ],
+      [ [ "water_clean", 1 ] ]
```

Comment:
Elaborating:  

Basic ice cream recipe includes cream (which isn't in game -- that's a possible opportunity for a dairy rework down the line, given that milking cows is a thing.  That + churns gives opportunities for butter, cheese, cream, et al crafting -- which is beyond the scope for this particular PR, but noting down).  

You -can- make ice cream in a bag, yes.  Maybe two ice cream recipes -- one with a bag (bag ice cream?) that has lower 'fun' involved (because it doesn't taste -quite- as good) and one that uses an ice cream maker that has higher fun?  Just spitballing ideas here.  That said, from memory, I know part of the ice cream making process involves actual ice (with a maker, you pack ice and rock salt around the container to help it freeze -- I think bag ice cream uses a very similar process).  

There's no ice in the game (other than frozen water) -- which we can get with a minifreezer.  Is there a way to require a specific -state- of something to cook?  In this case water (frozen)?  I'm guessing in the recipe for ice cream that water (clean) is a substitute for water (frozen) -- I'd also think that -any- water could work if that's the case.